contrib/opencensus-ext-azure/tests/test_azure_log_exporter.py

```diff
@@ -259,3 +261,194 @@ def test_log_record_not_sampled(self, requests_mock):
+            instrumentation_key='12345678-1234-5678-abcd-12345678abcd',
+            storage_path=os.path.join(TEST_FOLDER, self.id()),
+        )
+        logger.addHandler(handler)
+        logger.warning('action', extra={
+            'custom_dimensions':
+                {
+                    'key_1': 'value_1',
+                    'key_2': 'value_2'
+                }
+            })
+        handler.close()
+        post_body = requests_mock.call_args_list[0][1]['data']
+        self.assertTrue('action' in post_body)
+        self.assertTrue('key_1' in post_body)
+        self.assertTrue('key_2' in post_body)
+
+    @mock.patch('requests.post', return_value=mock.Mock())
+    def test_log_with_invalid_custom_properties(self, requests_mock):
+        logger = logging.getLogger(self.id())
+        handler = log_exporter.AzureEventHandler(
+            instrumentation_key='12345678-1234-5678-abcd-12345678abcd',
+            storage_path=os.path.join(TEST_FOLDER, self.id()),
+        )
+        logger.addHandler(handler)
+        logger.warning('action_1_%s', None)
+        logger.warning('action_2_%s', 'arg', extra={
+            'custom_dimensions': 'not_a_dict'
+        })
+        logger.warning('action_3_%s', 'arg', extra={
+            'notcustom_dimensions': {'key_1': 'value_1'}
+        })
+
+        handler.close()
+        self.assertEqual(len(os.listdir(handler.storage.path)), 0)
+        post_body = requests_mock.call_args_list[0][1]['data']
+        self.assertTrue('action_1_' in post_body)
+        self.assertTrue('action_2_arg' in post_body)
+        self.assertTrue('action_3_arg' in post_body)
+
+        self.assertFalse('not_a_dict' in post_body)
+        self.assertFalse('key_1' in post_body)
+
+    @mock.patch('requests.post', return_value=mock.Mock())
+    def test_log_record_sampled(self, requests_mock):
+        logger = logging.getLogger(self.id())
+        handler = log_exporter.AzureEventHandler(
+            instrumentation_key='12345678-1234-5678-abcd-12345678abcd',
+            logging_sampling_rate=1.0,
+        )
+        logger.addHandler(handler)
+        logger.warning('Hello_World')
+        logger.warning('Hello_World2')
+        logger.warning('Hello_World3')
+        logger.warning('Hello_World4')
+        handler.close()
+        post_body = requests_mock.call_args_list[0][1]['data']
+        self.assertTrue('Hello_World' in post_body)
+        self.assertTrue('Hello_World2' in post_body)
+        self.assertTrue('Hello_World3' in post_body)
+        self.assertTrue('Hello_World4' in post_body)
+
+    @mock.patch('requests.post', return_value=mock.Mock())
+    def test_log_record_not_sampled(self, requests_mock):
+        logger = logging.getLogger(self.id())
+        handler = log_exporter.AzureEventHandler(
+            instrumentation_key='12345678-1234-5678-abcd-12345678abcd',
+            logging_sampling_rate=0.0,
+        )
+        logger.addHandler(handler)
```

Comment:
nit could stick this in a contextmanager since it appears a few times and (I guess) you need to remember `handler.close()`